livekit-agents/livekit/agents/voice_assistant/voice_assistant.py

```diff
@@ -567,9 +567,17 @@ def _commit_user_question_if_needed() -> None:
             call_ctx = AssistantCallContext(self, speech_info.source)
             tk = _CallContextVar.set(call_ctx)
             self.emit("function_calls_collected", speech_info.source.function_calls)
-            called_fncs = speech_info.source.execute_functions()
-            tasks = [called_fnc.task for called_fnc in called_fncs]
-            await asyncio.gather(*tasks, return_exceptions=True)
+            called_fncs_info = speech_info.source.function_calls
+
+            called_fncs = []
+            for fnc in called_fncs_info:
+                called_fnc = fnc.execute()
+                called_fncs.append(called_fnc)
+                try:
+                    await called_fnc.task
+                except Exception:
```

Comment:
Should we log something here?

Reply:
no, the ai callables errors will be forwarded to the LLM.
But I think we should create an AIError type, so we can still log unexpected errors

Reply:
It might still make sense to log something no? When debugging an old session it could be useful

Reply:
we didn't before, I think this should be handled in a new PR, because we need to introduce a new special exception type that are marked as "expected"